### PR TITLE
[GLUTEN-2626] PlanJson load in debug level in WholeStageTransformer

### PR DESCRIPTION
**Pull request by loneylee**
## What changes were proposed in this pull request?

PlanJson not init all times. Only load in debug level.

## How was this patch tested?

Test by ut.

### CODE REVIEW

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI

**Comment by PHILO-HE** [comment]
Hi @loneylee, the CI reported test failure is just ignored by a commit yesterday. You can get your PR merged if only this test failed. Thanks!

**Review by PHILO-HE — APPROVED** [review]
Just corrected a typo in title. Looks great!

**Comment by marin-ma** [comment]
LGTM. Could you please do rebase?

File: gluten-core/src/main/scala/io/glutenproject/execution/WholeStageTransformer.scala [hunk 1]
```diff
@@ -295,7 +298,10 @@ case class WholeStageTransformer(child: SparkPlan)(val transformStageId: Int)
       val startTime = System.nanoTime()
       val resCtx = doWholeStageTransform()
 
-      logOnLevel(substraitPlanLogLevel, s"Generating substrait plan:\n$planJson")
+      if (log.isDebugEnabled()) {
+        logOnLevel(substraitPlanLogLevel, s"Generating substrait plan:\n$planJson")
+      }
+
```
Comment (zzcclp) [inline gluten-core/src/main/scala/io/glutenproject/execution/WholeStageTransformer.scala:304]: remove these code, planJson already is included when executing `generateTreeString`.

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI

File: gluten-core/src/main/scala/io/glutenproject/execution/WholeStageTransformer.scala (hunk 1 above)
Comment (loneylee) [reply gluten-core/src/main/scala/io/glutenproject/execution/WholeStageTransformer.scala]: fixed

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**